api/src/Consumer/Handler/BuildArchiveHandler.php

```diff
@@ -4,47 +4,45 @@
 
 namespace App\Consumer\Handler;
 
-use App\Archive\ArchiveManager;
+use Throwable;
 use App\Entity\Archive;
-use Arthem\Bundle\RabbitBundle\Consumer\Event\AbstractEntityManagerHandler;
-use Arthem\Bundle\RabbitBundle\Consumer\Event\EventMessage;
-use Arthem\Bundle\RabbitBundle\Consumer\Exception\ObjectNotFoundForHandlerException;
 use Doctrine\DBAL\LockMode;
-use Doctrine\ORM\EntityManager;
-use Throwable;
+use App\Archive\ArchiveManager;
+use App\Consumer\Handler\BuildMessage;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\Messenger\Attribute\AsMessageHandler;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
 
-class BuildArchiveHandler extends AbstractEntityManagerHandler
+#[AsMessageHandler]
+class BuildArchiveHandler
 {
-    const EVENT = 'build_archive';
-
     private ArchiveManager $archiveManager;
+    private EntityManagerInterface $em;
 
-    public function __construct(ArchiveManager $archiveManager)
+    public function __construct(ArchiveManager $archiveManager, EntityManagerInterface $em)
     {
         $this->archiveManager = $archiveManager;
+        $this->em = $em;
     }
 
-    public function handle(EventMessage $message): void
+    public function __invoke(BuildMessage $message): void
```

Comment:
Please keep the same name as the handler:

```suggestion
    public function __invoke(BuildArchive $message): void
```